server/neptune/workflows/internal/deploy/revision/queue/queue.go

```diff
@@ -19,28 +25,39 @@ type Deploy struct {
 	queue *priority
 
 	// mutable: default is unlocked
-	lock LockStatus
+	lock               LockState
+	lockStatusCallback func(workflow.Context, *Deploy)
```

Comment:
is this mutable?

Reply:
nope ill move this out from under this block